src/Routing/RedirectableUrlMatcher.php

```diff
@@ -0,0 +1,102 @@
+<?php
+
+/*
+ * This file is part of the Symfony CMF package.
+ *
+ * (c) Symfony CMF
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Cmf\Bundle\RoutingBundle\Routing;
+
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\Routing\Exception\ResourceNotFoundException;
+use Symfony\Component\Routing\Generator\UrlGeneratorInterface;
+use Symfony\Component\Routing\Matcher\RedirectableUrlMatcherInterface;
+use Symfony\Component\Routing\Matcher\RequestMatcherInterface;
+use Symfony\Component\Routing\RequestContext;
+
+class RedirectableUrlMatcher implements RequestMatcherInterface, UrlGeneratorInterface, RedirectableUrlMatcherInterface
+{
+    /**
+     * @var RequestMatcherInterface|UrlGeneratorInterface
+     */
+    private RequestMatcherInterface $router;
+
+    public function __construct(RequestMatcherInterface $router)
```

Comment:
We could type hint RouterInterface here but this just add functions to copy paste...